File: src/23.ts
import { getEnglandArticle } from './share/get-england-article'
import { notEmpty } from './helper/iterable-helper'

const sectionRegex = /(?<open>==+) *(?<name>[^= ]+) *(?<close>==+)/
console.log(
    getEnglandArticle()
        .split('\n')
        .map(row => row.match(sectionRegex))
        .filter(notEmpty)
        .map(match => match.groups)
        .filter(notEmpty)
        .map(({ name, open }) => ({
            name,
            level: open.length - 1,
        }))
)

File: src/23.py
import re

from share.get_england_text import get_england_text

text = get_england_text()
section_regex = re.compile(r"(?P<open>==+) *(?P<name>[^= ]+) *(?P=open)")
print(
    "\n".join(
        map(
            lambda section: f"{section[1]} {len(section[0]) - 1}",
            section_regex.findall(text),
        )
    )
)
